knowledge-base/pdfviewer-sign-pdfs.md

```diff
@@ -0,0 +1,152 @@
+		var signatureField = new SignatureField(signatureName);
+		signatureField.Signature = new Telerik.Windows.Documents.Fixed.Model.DigitalSignatures.Signature(certificate);
+
+        // close the local cert store now that we're done
+        x509Store.Close();
+
+
+        // **** PHASE 3 - IMPORT DOCUMENT and insert signature field **** //
+
+        var document = new PdfFormatProvider().Import(unsignedDocumentBytes);
+
+        // For the sake of this demo, we are adding a new page and adding a new signature field there, if your document already has a signature field, you can search for it instead
+        RadFixedPage page = document.Pages.AddPage();
+
+        // This is the Form XObject element that represents the contents of the signature field.
+        var form = new Telerik.Windows.Documents.Fixed.Model.Objects.Form
+        {
+            FormSource = new FormSource { Size = new Size(220, 220) }
+        };
+
+        // We will use the editor to fill the Form XObject.
+        var formEditor = new FixedContentEditor(form.FormSource);
+        form.Position.Translate(10, 10);
+        formEditor.DrawCircle(new Point(50, 50), 20);
+        formEditor.DrawText(signatureName);
+
+		// The widget contains the Form XObject and defines the appearance of the signature field.
+		var widget = signatureField.Widgets.AddWidget();
+		widget.Rect = new Rect(200, 600, 100, 100);
+		widget.Border = new AnnotationBorder(10, AnnotationBorderStyle.Solid, null);
+		widget.Content.NormalContentSource = form.FormSource;
+		widget.RecalculateContent();
+
+        // Finally add the SignatureWidget to the page's annotations
+		page.Annotations.Add(widget);
+
+		var editor = new FixedContentEditor(page);
+		editor.Position.Translate(200, 400);
+		editor.DrawForm(form.FormSource);
+		document.AcroForm.FormFields.Add(signatureField);
+		widget.RecalculateContent();
+		widget.AppearanceCharacteristics.Background = new Telerik.Windows.Documents.Fixed.Model.ColorSpaces.RgbColor(255, 0, 0);
+
+        // **** PHASE 4 - EXPORT DOCUMENT **** //
+        byte[] signedDocBytes = new PdfFormatProvider().Export(document);
+
+        return signedDocBytes;
+    }
+}
+```
+
+## Notes
+
+- The Telerik Blazor PdfViewer does not currently support digital signature management in the UI.
+- You can use Telerik PdfProcessing to digitally sign the PDF on the server-side.
+- The web application cannot access the user's guest operating system's X509 Certificate Store due to security restrictions in modern web browsers. Your application will only have access to X509 certificates installed on the server side. 
+
+## See Also
+
+- [PdfViewer - Overview](https://docs.telerik.com/blazor-ui/components/pdfviewer/overview)
+- [Document Processing Digital Signature documentation](https://docs.telerik.com/devtools/document-processing/libraries/radpdfprocessing/features/security/digital-signatures)
```

Comment:
Double-check the link - it's a 404 for me. https://docs.telerik.com/devtools/document-processing/libraries/radpdfprocessing/features/digital-signature seems to be the correct link.

Reply:
You're correct, it is in the Features section of the documentation. I'm not sure  where this one came from, could be hallucination, or that it is a last minute copy/paste mistake on my part.